src/sinks/opentelemetry_metrics/service.rs

```diff
@@ -0,0 +1,243 @@
+                HttpStatusRetryLogic::new(|resp: &http::Response<Body>| resp.status()),
+                service,
+                buffer,
+                batch.timeout,
+            )
+            .sink_map_err(
+                |error| error!(message = "Fatal OpentelemetryMetrics sink error.", %error),
+            )
+            .with_flat_map(move |event: Event| {
+                stream::iter({
+                    let byte_size = event.allocated_bytes();
+                    let json_byte_size = event.estimated_json_encoded_size_of();
+                    normalizer.normalize(event.into_metric()).map(|mut metric| {
+                        let namespace = metric
+                            .take_namespace()
+                            .unwrap_or_else(|| default_namespace.clone());
+                        Ok(EncodedEvent::new(
+                            PartitionInnerBuffer::new(metric, namespace),
+                            byte_size,
+                            json_byte_size,
+                        ))
+                    })
+                })
+            });
+
+        #[allow(deprecated)]
+        Ok(VectorSink::from_event_sink(sink))
+    }
+}
+
+impl Service<PartitionInnerBuffer<Vec<Metric>, String>> for OpentelemetryMetricsSvc {
+    type Response = http::Response<Body>;
+    type Error = crate::Error;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, items: PartitionInnerBuffer<Vec<Metric>, String>) -> Self::Future {
+        let (items, namespace) = items.into_parts();
+        let metrics = encode_metrics(items, self.config.aggregation_temporality);
+        if metrics.is_empty() {
+            return Box::pin(future::ok(
+                http::Response::builder()
+                    .status(StatusCode::OK)
+                    .body(Body::empty())
+                    .unwrap(),
+            ));
+        }
+
+        let client = self.client.clone();
+        let namespace_clone = namespace.clone();
+        let endpoint = self.config.endpoint.clone();
+
+        Box::pin(async move {
+            // Create the ResourceMetrics structure
+            let resource_metrics = ResourceMetrics {
+                resource: Some(Resource {
+                    attributes: vec![KeyValue {
+                        key: "service.name".to_string(),
+                        value: Some(AnyValue {
+                            value: Some(any_value::Value::StringValue(namespace)),
+                        }),
+                    }],
+                    dropped_attributes_count: 0,
+                }),
+                scope_metrics: vec![ScopeMetrics {
+                    scope: Some(InstrumentationScope {
+                        name: "vector".to_string(),
+                        version: env!("CARGO_PKG_VERSION").to_string(),
```

Comment:
Not sure if this context should come from the metric event's tags or the sending agent (Vector).